test/miner-test.js

```diff
@@ -180,4 +181,81 @@ describe('Miner', function() {
       assert(txids.has(block.txs[i].hash()));
     }
   });
+
+   it('should not include free transaction in a block', async () => {
+    // Miner does not have any room for free TXs
+    miner.options.minWeight = 0;
+
+    const value = 1 * 1e6;
+    const fee = 0;
+
+    // Get a change address
+    const change = wallet.createChange().getAddress();
+    const mtx = new MTX();
+    coin = wallet.getCoins()[0];
+    mtx.addCoin(coin);
+    mtx.addOutput(addr, value);
+    mtx.addOutput(change, coin.value - value - fee); // no fee
+    wallet.sign(mtx);
+    parentTX = mtx.toTX();
+
+    await mempool.addTX(parentTX, -1);
+    assert.strictEqual(mempool.map.size, 1);
+
+    const block = await miner.mineBlock(chain.tip, addr);
+    await chain.add(block);
+
+    // TX is still in mempool, nothing in block except coinbase
+    assert.strictEqual(mempool.map.size, 1);
```

Comment:
I think this test block should begin either with a `mempool.reset()` or assertion that mempool is empty. What might be even better is starting a new `describe` block called `CPFP` or something. Especially since the "should not include free transaction in a block" test is basically a duplicate. We know we can't include free TXs in blocks from L127-L165.

Reply:
yea I think a separate describe block is better, since this itself got way too long for just miner test. or maybe a complete different test file for CPFP and in future RBF (and other fee bumping mechanisms) might be better.